react/hooks/useScrollDirection.js

```diff
@@ -0,0 +1,23 @@
+import { useState, useLayoutEffect } from 'react'
+import useScroll from './useScroll'
+
+/**
+ * Hook that handles the scroll direction
+ * @returns {Number, Boolean, Boolean} - the current scroll value and if the scroll is up, or down
+ */
+const useScrollDirection = () => {
+  const { scroll } = useScroll()
+  const [scrollingUp, setScrollingUp] = useState(false)
+  const [scrollingDown, setScrollingDown] = useState(false)
+  const [lastScroll, setLastScroll] = useState(0)
+
+  useLayoutEffect(() => {
+    setLastScroll(scroll)
+    setScrollingUp(lastScroll > scroll)
+    setScrollingDown(lastScroll < scroll)
```

Comment:
this is related to the behaviour rather than to the code itself, but:

 a threshold could be set here, to prevent the menu from flickering if the user changes the scroll by 1 px or so (which happens on finer-grained scrolling such as touch devices or macbook touchpads)

another thing is that perhaps it's not ideal to have two booleans, `scrollingUp` and `scrollingDown`, opposite of each other—it's a bit redundant and may cause issues if this code becomes a bit more complex.

Reply:
Done ✅